pandas/io/packers.py

```diff
@@ -255,17 +261,36 @@ def unconvert(values, dtype, compress=None):
     if not as_is_ext:
         values = values.encode('latin1')
 
-    if compress == u'zlib':
-        import zlib
-        values = zlib.decompress(values)
-        return np.frombuffer(values, dtype=dtype)
-
-    elif compress == u'blosc':
-        import blosc
-        values = blosc.decompress(values)
-        return np.frombuffer(values, dtype=dtype)
+    if compress:
+        if compress == u'zlib':
+            from zlib import decompress
+        elif compress == u'blosc':
+            from blosc import decompress
+        else:
+            raise ValueError("compress must be one of 'zlib' or 'blosc'")
+
+        values = decompress(values)
+        if sys.getrefcount(values) == 2:
```

Comment:
The issue ius that this we will be working with memory allocated for a `bytes`. For example:

```python
In [1]: buf = bytes(range(4))

In [2]: np.frombuffer(buf, dtype='uint8')
Out[2]: array([0, 1, 2, 3], dtype=uint8)

In [3]: _2.flags.writeable = True

In [4]: _2[0] = 5

In [5]: buf
Out[5]: b'\x05\x01\x02\x03'
```

The only case where we can just use the buffer is when we are the sole owner of values and no one can reference `values` again. In the other cases we need to just copy the data so we can mutate while preserving normal python semantics of immutable strings.